feat: Add azure sql Db module

HadwaAbdelhalem opened the pull request:
Implements basic example for Azure SQL Server and Database 

Closes #610

rguthriemsft commented on modules/azure/sql.go:
```diff
@@ -0,0 +1,185 @@
+}
+
+// GetDatabaseStatus is a helper function that gets the sql db state
+func GetDatabaseStatus(t testing.TestingT, resGroupName string, serverName string, dbName string, subscriptionID string) string {
```
Add a method to get the SQL Server IP address.

HadwaAbdelhalem replied:
Azure sql servers have no static iIP but a dynamic IP that changes all the time. thus connections to them rely on using the full domain name.  there is a method that get the full domain name for sql server in the module. Also looking at the properties of the server properties , it has no way to check/retrieve the current server IP

rguthriemsft commented on modules/azure/sql.go:
```diff
@@ -0,0 +1,185 @@
+
+	//Return DB status
+	return *sqlDb.Status, nil
+}
```
Add a method to get a list of databases on the sql server.

HadwaAbdelhalem replied:
added ListSQLServerDatabasesE

rguthriemsft reviewed (changes requested):
Take a look at the two methods I suggested you add.  Ping me if you want to discuss.

HadwaAbdelhalem commented on examples/azure/terraform-azure-sqldb-example/variables.tf:
```diff
@@ -0,0 +1,43 @@
+variable sqlserver_admin_login {
+    description = "The administrator login name for the sql server."
+    type        = string
+    default     = "3cx1n1z5r079b"
```
Administrator is not a valid login name in this version of SQL Server, using it will cause the create sql server API to return 400-InvalidLoginName error https://docs.microsoft.com/en-us/rest/api/sql/managedinstances/update

rguthriemsft replied:
ah, we should just make it something a little more meaningful.

HadwaAbdelhalem replied:
fixed

HadwaAbdelhalem commented on examples/azure/terraform-azure-sqldb-example/outputs.tf:
```diff
@@ -0,0 +1,32 @@
+  sensitive = true
+}
+
+output "resource_group_name" {
```
fixed

HadwaAbdelhalem commented on examples/azure/terraform-azure-sqldb-example/main.tf:
```diff
@@ -0,0 +1,61 @@
+# See test/terraform_azure_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_resource_group" "sql" {
```
fixed

HadwaAbdelhalem commented on examples/azure/terraform-azure-sqldb-example/main.tf:
```diff
@@ -0,0 +1,61 @@
+# ------------------------------------------------------------------------------
+
+provider "azurerm" {
+  version = "~>2.8.0"
```
fixed

rguthriemsft commented on examples/azure/terraform-azure-sqldb-example/main.tf:
```diff
@@ -0,0 +1,61 @@
+  tags = {
+    environment = var.tags
+  }
+}
```
add newline

HadwaAbdelhalem replied:
fixed

rguthriemsft commented on examples/azure/terraform-azure-sqldb-example/outputs.tf:
```diff
@@ -0,0 +1,32 @@
+output "sql_server_admin_login_pass" {
+  value     = azurerm_sql_server.sqlserver.administrator_login_password
+  sensitive = true
+}
```
add newline

HadwaAbdelhalem replied:
fixed

rguthriemsft commented on examples/azure/terraform-azure-sqldb-example/variables.tf:
```diff
@@ -0,0 +1,43 @@
+  description = "A postfix string to centrally mitigate resource name collisions"
+  type        = string
+  default     = "resource"
+}
```
add newline

HadwaAbdelhalem replied:
fixed

rguthriemsft reviewed (changes requested):
@HadwaAbdelhalem please see comments

rguthriemsft commented:
Passing CI link https://github.com/HadwaAbdelhalem/terratest/actions/runs/283170487

yorinasub17 commented on modules/azure/sql.go:
```diff
@@ -0,0 +1,211 @@
+package azure
```
NIT: I mentioned this in another PR ([this comment](https://github.com/gruntwork-io/terratest/pull/641#discussion_r498890874)), but it feels extraneous to provide getter functions that go directly to the attribute of an object, and instead we should provide a single function that returns the object.

For example, instead of offering `GetSQLServerID` and `GetSQLServerFullDomainName`, we could offer `GetSQLServer` which returns the object and one can access the `ID` and `FullyQualifiedDomainName` fields on the same object.

HadwaAbdelhalem replied:
fixed, created two getter one for sqlserver and one for sqldb and removed the properties getters

yorinasub17 reviewed (commented):
Mostly LGTM! Had some style nits, but did have one question about consolidating the functions.

HadwaAbdelhalem commented on examples/azure/terraform-azure-sqldb-example/main.tf:
```diff
@@ -0,0 +1,61 @@
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# See test/terraform_azure_example_test.go for how to write automated tests for this code.
```
fixed

HadwaAbdelhalem commented on examples/azure/terraform-azure-sqldb-example/main.tf:
```diff
@@ -0,0 +1,61 @@
+
+# ------------------------------------------------------------------------------
+# CONFIGURE OUR AZURE CONNECTION
+# ------------------------------------------------------------------------------
```
fixed

HadwaAbdelhalem commented on modules/azure/sql.go:
```diff
@@ -0,0 +1,211 @@
+	return &sqlClient, nil
+}
+
+// GetSQLServerID is a helper function that gets the sql server ID
```
fixed

HadwaAbdelhalem commented on examples/azure/terraform-azure-sqldb-example/README.md:
```diff
@@ -0,0 +1,51 @@
+# Terraform Azure SQL DB Example
```
fixed

HadwaAbdelhalem commented:
@yorinasub17  addressed comments. Here is a link to a passed CI execution https://github.com/HadwaAbdelhalem/terratest/runs/1201007991?check_suite_focus=true

rguthriemsft reviewed (approved):
Look good to me

yorinasub17 reviewed (approved):
LGTM!
